rest/model/subscriber.go

```diff
@@ -0,0 +1,149 @@
+package model
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen/model/event"
+)
+
+type APISubscriber struct {
+	Type   APIString   `json:"type"`
+	Target interface{} `json:"target"`
+}
+
+type APIGithubPRSubscriber struct {
+	Owner    APIString `json:"owner"`
+	Repo     APIString `json:"repo"`
+	PRNumber int       `json:"pr_number"`
+	Ref      APIString `json:"ref"`
+}
+
+type APIWebhookSubscriber struct {
+	URL    APIString `json:"url"`
+	Secret APIString `json:"secret"`
+}
+
+func (s *APISubscriber) BuildFromService(h interface{}) error {
+	switch v := h.(type) {
+
+	case event.Subscriber:
+		s.Type = ToAPIString(v.Type)
+		var target interface{}
+
+		switch v.Type {
+
+		case event.GithubPullRequestSubscriberType:
+			sub := APIGithubPRSubscriber{}
+			err := sub.BuildFromService(v.Target)
+			if err != nil {
+				return err
+			}
+			target = sub
+
+		case event.EvergreenWebhookSubscriberType:
+			sub := APIWebhookSubscriber{}
+			err := sub.BuildFromService(v.Target)
+			if err != nil {
+				return err
+			}
+			target = sub
+
+		case event.JIRAIssueSubscriberType, event.JIRACommentSubscriberType,
+			event.EmailSubscriberType, event.SlackSubscriberType:
+			target = v.Target
+
+		default:
+			return fmt.Errorf("unknown subscriber type: '%s'", v.Type)
+		}
+
+		s.Target = target
+
+	default:
+		return errors.New("unknown type for APISubscriber")
+	}
+
+	return nil
+}
+
+func (s *APISubscriber) ToService() (interface{}, error) {
+	var target interface{}
+	var err error
+	out := event.Subscriber{
+		Type: FromAPIString(s.Type),
+	}
+	switch FromAPIString(s.Type) {
+
+	case event.GithubPullRequestSubscriberType:
+		apiModel, ok := s.Target.(APIGithubPRSubscriber)
+		if !ok {
+			return nil, errors.New("unable to convert to APIGithubPRSubscriber")
```

Comment:
I think it makes sense to have this return `*rest.APIError`s with BadRequest in them. Thoughts?